generate_homepage_from_content.py

```diff
@@ -0,0 +1,215 @@
+    appropriate HTML template."""
+    website_content = WEBSITE_CONTENT_TEMPLATE % (quote, url)
+    return CONTENT_CONTAINER % ('website', website_content)
+
+
+def html_element_from_content(content_dict):
+    """Given a dict representing content (a type, a url, and optionally a
+    quote), returns the content wrapped in the appropriate HTML element
+    (ready for insertion on the homepage)."""
+    content_type = content_dict.get(CSV_KEY_TYPE)
+    if not content_type:
+        print 'No content type provided for entry: "%s". Skipping.' % content_dict
+        return
+    elif content_type not in CONTENT_TYPES:
+        print ('Unrecognized content type ("%s") in entry: "%s". Skipping.'
+               % (content_type, content_dict))
+        return
+    elif content_type in EMBEDDED_CONTENT_TYPES:
+        url = content_dict.get(CSV_KEY_URL)
+        if not url:
+            print 'No url provided for entry: "%s". Skipping.' % content_dict
+            return
+        elem = html_element_from_embedded_content(url, content_type)
+    elif content_type == CONTENT_TYPE_WEBSITE:
+        url = content_dict.get(CSV_KEY_URL)
+        quote = content_dict.get(CSV_KEY_QUOTE)
+        elem = html_element_from_website_content(url, quote)
+
+    # make sure to escape any lone '%' signs
+    return elem.replace('%', '%%')
+
+
+def content_from_csv(filepath):
+    """Given the filepath of a csv containing content, returns a list of dicts,
+    each dict representing a row."""
+    return list(csv.DictReader(open(filepath)))
+
+
+def validate_filepath(filepath):
+    """Make sure given filepath is a valid file."""
+    if not os.path.isfile(filepath):
+        print ('ERROR: Filepath provided ("%s") is not a file (may be a '
+               'directory or an invalid path).' % filepath)
+        sys.exit(1)
+    return
+
+
+def main():
+    args = parse_command_line_args()
+
+    # set filepaths from command line args or default, validate filepaths.
+    if args.content_csv:
+        content_csv_filepath = args.content_csv
+    else:
+        print 'No content csv filepath provided, using default: %s' % DEFAULT_CONTENT_CSV_PATH
+        content_csv_filepath = DEFAULT_CONTENT_CSV_PATH
+    validate_filepath(content_csv_filepath)
+
+    if args.page_template:
+        page_template_filepath = args.page_template
+    else:
+        print 'No page template filepath provided, using default: %s' % DEFAULT_PAGE_TEMPLATE_PATH
+        page_template_filepath = DEFAULT_PAGE_TEMPLATE_PATH
+    validate_filepath(page_template_filepath)
+
+    if args.outfile:
+        outfile_filepath = args.outfile
+    else:
+        print 'No outfile filepath provided, using default: %s' % DEFAULT_OUTFILE_PATH
+        outfile_filepath = DEFAULT_OUTFILE_PATH
+
+    print '-'*80
+
+    # read in content csv, generate html elements for each piece of content
+    # (getting embed code from API where appropriate.)
+    content = content_from_csv(content_csv_filepath)
```

Comment:
could also just iterate over the rows using the DictReader, rather than making a list and then iterating over it: 
>>> with open('names.csv') as csvfile:
...     reader = csv.DictReader(csvfile)
...     for row in reader: